workbench-core/base/src/utilities/queryParamsFiltersUtils.ts

```diff
@@ -0,0 +1,113 @@
+  if (filter && sort) {
+    throw Boom.badRequest('Cannot apply a filter and sort at the same time');
+  }
+  const filterAttributesLength = filter ? Object.keys(filter).length : 0;
+  // Check that at most one filter is defined because we not support more than one filter
+  if (filterAttributesLength > 1) {
+    throw Boom.badRequest('Cannot apply more than one filter.');
+  }
+  // Check that at most one sort attribute is defined because we not support sorting by more than one attribute
+  if (sort && Object.keys(sort).length > 1) {
+    throw Boom.badRequest('Cannot sort by more than one attribute.');
+  }
+}
+
+/************************************************************
+ * Parses Object with format \{ field: \{ operator: value \} \} into QueryParam object consumed by dynamoDB
+ *
+ * @param filter - filter request object field
+ * @param sortKey - field name in dynamoDB
+ * @param gsi - gsi to use for filter
+ *
+ * @returns QueryParams object
+ ************************************************************/
+export function parseQueryParamFilter<T>(
+  filter: QueryParameterFilter<T> | undefined,
+  sortKey: string,
+  gsi: string
+): QueryParams | undefined {
+  if (!filter || Object.keys(filter).length === 0 || Object.keys(filter).length > 1) {
+    return undefined;
+  }
+  if (filter?.between && filter.between.value1 && filter.between.value2) {
+    const filterProperty = getFilterPropertyByType(filter.between.value1);
+    const value1 = { [filterProperty]: filter.between.value1 } as unknown as AttributeValue;
+    const value2 = { [filterProperty]: filter.between.value2 } as unknown as AttributeValue;
+    return {
+      index: gsi,
+      sortKey: sortKey,
+      between: { value1, value2 }
+    };
+  }
+  const filterOperator: keyof QueryParameterFilter<T> = Object.keys(
+    filter
+  )[0] as keyof QueryParameterFilter<T>;
+  //eslint-disable-next-line security/detect-object-injection
+  if (filterOperator && filter[filterOperator]) {
+    //eslint-disable-next-line security/detect-object-injection
+    const propertyType = getFilterPropertyByType(filter[filterOperator]);
+    //eslint-disable-next-line security/detect-object-injection
+    const value = { [propertyType]: filter[filterOperator] } as unknown as AttributeValue;
+    return {
+      index: gsi,
+      sortKey: sortKey,
+      [filterOperator]: { ...value }
+    };
+  }
+}
+
+/************************************************************
+ * Creates a sort object into a QueryParam object to be consumed by dynamoDB
+ * @param sort - type of sorting
+ * @param sortKey - field name in dynamoDB
+ * @param gsi - gsi to use for sort
+ *
+ * @returns QueryParams object
+ ************************************************************/
+export function parseQueryParamSort(
+  sort: 'asc' | 'desc' | undefined,
+  sortKey: string,
+  gsi: string
+): QueryParams | undefined {
+  if (!sort) {
+    return undefined;
+  }
+  const forward: boolean = sort !== 'desc';
```

Comment:
Once the TS is transpiled to JS, would it be possible for someone to send in a value that wasn't `asc`, `desc`, or `undefined` via a query param? I'm worried that only checking for `desc` means that at runtime someone could still send in something invalid.

Reply:
the api is able to limit those values with the scema, on this layer I will just asume that if not asc or desc default is asc in the new commit I will add